Diary/Diary/DiaryService.cs

```diff
@@ -0,0 +1,220 @@
+                    break;
+
+                case 2:
+                    Records2Parametrs(taskParametrs, numberTask);
+                    break;
+
+                case 1:
+                    Records1Parametr(taskParametrs, numberTask);
+                    break;
+
+                default:
+                    Console.WriteLine("incorrect parameter format");
+                    break;
+            }
+        }
+
+        private static void Records1Parametr(string[] taskParametrs, int numberTask)
+        {
+            PriorityTask diaryTask = new(taskParametrs[0]);
+            weeklyTasks[numberTask] = diaryTask;
+        }
+
+        private static void Records2Parametrs(string[] taskParametrs, int numberTask)
+        {
+            PriorityTask diaryTask = new(taskParametrs[0],
+                                                   (taskParametrs[1]));
+            weeklyTasks[numberTask] = diaryTask;
+
+        }
+
+        private static void Records3Parametrs(string[] taskParametrs, int numberTask)
+        {
+            PriorityTask diaryTask = new(taskParametrs[0],
+                                                      DateTime.Parse(taskParametrs[1]),
+                                                      DateTime.Parse(taskParametrs[2]));
+            weeklyTasks[numberTask] = diaryTask;
+            diaryTask.GetAlarm();
+        }
+
+        private static void Record4Parametrs(string[] taskParametrs, int numberTask)
+        {
+            PriorityTask diaryTask = new(taskParametrs[0],
+                                                                          DateTime.Parse(taskParametrs[1]),
+                                                                          DateTime.Parse(taskParametrs[2]),
+                                                                          taskParametrs[3]);
+            weeklyTasks[numberTask] = diaryTask;
+            diaryTask.GetAlarm();
+        }
+
+
+        private static string CheckTime(int i)
+        {
+            if (weeklyTasks[i - 1].timeTask == default(DateTime))
+            {
+                return " ";
+            }
+            return weeklyTasks[i - 1].timeTask.ToLongTimeString();
+        }
+
+        private static string CheckData(int i)
+        {
+            if (weeklyTasks[i - 1].dataTask == default(DateTime))
+            {
+                return " ";
+            }
+            return weeklyTasks[i - 1].dataTask.ToShortDateString();
+        }
+
+
+        private void FiltrToPriority(int numberTask)
+        {
+            Console.WriteLine("select priority");
+            string filtr = Console.ReadLine();
+            PriorityTask[] filtrTask = new PriorityTask[numberTask];
+            var i = 0;
+            var x = 0;
+
+            do
+            {
+                if (weeklyTasks[i].priority == filtr)
+                {
+                    filtrTask[x] = weeklyTasks[i];
+                    x++;
+                }
+                i++;
```

Comment:
too many iterators IMO, you can show tasks one by one or use advanced collections like `List` to collect the result.